### PR TITLE
Trac2869 part2

### PR DESCRIPTION
**Pull request by olgamelnichuk**
@alf239 - please, review

### CODE REVIEW

File: atlas-web/src/main/java/ae3/service/experiment/AtlasExperimentAnalyticsViewService.java [hunk 1]
```diff
@@ -28,9 +29,29 @@ import java.util.*;
 
 public class AtlasExperimentAnalyticsViewService {
 
-    final private Logger log = LoggerFactory.getLogger(getClass());
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private static final Function<UpDownCondition, String> SUPPORTED_UP_DOWN = new Function<UpDownCondition, String>() {
+        @Override
+        public String apply(@Nullable UpDownCondition input) {
+            switch (input) {
```
Comment (alf239) [inline atlas-web/src/main/java/ae3/service/experiment/AtlasExperimentAnalyticsViewService.java]: EnumMap?

File: atlas-web/src/main/java/ae3/service/experiment/rcommand/RCommandResult.java [hunk 2]
```diff
@@ -15,19 +15,23 @@ public class RCommandResult {
     }
 
     public int[] getIntValues(String valueName) {
-        return ret((RInteger) getValue(valueName));
```
Comment (alf239) [inline atlas-web/src/main/java/ae3/service/experiment/rcommand/RCommandResult.java:18]: Don't we have empty sets in R? Are we sure R returns null for empty sets?

File: atlas-web/src/main/java/ae3/service/experiment/AtlasExperimentAnalyticsViewService.java (hunk 1 above)
Comment (alf239) [inline atlas-web/src/main/java/ae3/service/experiment/AtlasExperimentAnalyticsViewService.java]: Or, better, enum

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java [hunk 3]
```diff
@@ -197,29 +187,46 @@ public class ApiQueryRequestHandler extends AbstractRestRequestHandler implement
...
+
+                                        NetCDFDescriptor ncdfDescr =
+                                                atlasNetCDFDAO.getNetCdfFile(experiment.getAccession(), netCDFProxyPredicate);
+
+                                        if (ncdfDescr != null) {
+
+                                            Collection<String> factors = Collections.emptyList();
+                                            Collection<String> factorValues = Collections.emptyList();
+                                            if (!conditions.isEmpty()) {
+                                                factors = Arrays.asList(conditions.iterator().next().getFactor());
```
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java:205]: I'm afraid it's not something user means. Create a Trac issue to fix/address this. Or, at least, document/fix the StructuredQuery if it only supports one factor

File: atlas-web/src/main/java/uk/ac/ebi/gxa/web/controller/ExperimentViewController.java [hunk 4]
```diff
@@ -145,6 +176,226 @@ public class ExperimentViewController extends ExperimentViewControllerBase {
...
+
+        List<Long> geneIds = findGeneIds(gid);
+
+        Predicate<NetCDFProxy> ncdfPredicate = alwaysTrue();
+        if (!geneIds.isEmpty()) {
+            ncdfPredicate = containsAtLeastOneGene(geneIds);
+        }
+
+        if (!isNullOrEmpty(adAcc)) {
+            ncdfPredicate = hasArrayDesign(adAcc);
```
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/web/controller/ExperimentViewController.java:215]: we're rewriting the previous choice; could we please combine ifs?

Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/web/controller/ExperimentViewController.java:215]: my mistake, by the way — my apologies :(

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java (hunk 3 above)
Comment (olgamelnichuk) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java:205]: Ticket #2954 Ambiguous behaviour of getting top 10 genes in the experiment API call

File: atlas-web/src/main/java/ae3/service/experiment/rcommand/RCommandResult.java (hunk 2 above)
Comment (alf239) [inline atlas-web/src/main/java/ae3/service/experiment/rcommand/RCommandResult.java:18]: Added notes to class's JavaDoc, added RCommandResultTest

**Pull request merged**